lib/features/thread/presentation/thread_controller.dart

```diff
@@ -455,7 +458,21 @@ class ThreadController extends BaseController {
 
   void _loadMoreEmailsSuccess(LoadMoreEmailsSuccess success) {
     if (success.emailList.isNotEmpty) {
-      emailList.addAll(success.emailList);
+      bool ableAppendLoadMore = true;
+      for (var email in success.emailList) {
```

Comment:
Why don't you use `list.where` to check the condition. IMO, Using the `for` loop takes quite a while.